agent/background.go

```diff
@@ -129,3 +131,22 @@ func (a *Agent) withCallbackTimeout(ctx context.Context, tc *taskContext) (conte
 	}
 	return context.WithTimeout(ctx, timeout)
 }
+
+func (a *Agent) startSpotTerminationWatcher(ctx context.Context, tc *taskContext, cancel context.CancelFunc) {
```

Comment:
tc isn't used. If you take it out, you can also move this into the main agent scope which for what it's doing at the moment, feels a bit more natural, but I don't feel strongly about that

Reply:
I removed `tc`. However, I don't think I can put it in the main agent loop (if that's what you mean by scope), since `Distro` is set in the `taskConfig`, which might not be non-nil until `runTask` sets it.